src/demo/data.js

```diff
@@ -34,19 +34,66 @@ const enrollments = [...Array(numEnrollments)].map((_, idx) => {
   };
 });
 
+const filteredEnrollments = (options) => {
+  let filterEnrollments = enrollments;
+  if (options.learner_activity) {
+    filterEnrollments = filterLearnerActivity(filterEnrollments, options.learner_activity);
+  }
+  if (options.passed_date) {
+    filterEnrollments = filterPastWeekCompletions(filterEnrollments);
+  }
+  return filterEnrollments;
+};
+
+const filterPastWeekCompletions = (enrollments) => {
+  let passedDate = faker.date.recent(7);
+  let today = faker.date.recent(0);
+  return enrollments.filter((enrollment) => {
+    return enrollment.has_passed &&
+            enrollment.passed_timestamp.getTime() >= passedDate.getTime() &&
+            enrollment.passed_timestamp.getTime() <= today.getTime();
+  });
+};
+
+const filterLearnerActivity = (enrollments, filterOption) => {
+  if (filterOption === 'active_past_week') {
+    let lastActivityDate = faker.date.recent(7);
```

Comment:
@mushtaqak This is your problem - you shouldn't use `faker.date.recent` as that will return a random date sometime in the last 7 days. Instead use `let weekAgo = new Date(); weekAgo.setDate(weekAgo.getDate() - 2);`

Reply:
Exactly, I found the issue as well. Couldn't push the code due to some other mess I had created 😌